src/clusterfuzz/_internal/tests/core/metrics/monitor_test.py

```diff
@@ -157,6 +158,47 @@ def test_cumulative_distribution_metric_fixed(self):
         2,
     ], result.buckets)
 
+  def test_gauge_metric_success(self):
+    """Test gauge metric success."""
+    self.assertIsInstance(
+        monitor.GaugeMetric('g', 'desc', field_spec=None), monitor._GaugeMetric)
+
+  def test_gauge_metric_failure(self):
+    """Test gauge metric failure."""
+    self.mock.check_module_loaded.return_value = False
+    gauge = monitor.GaugeMetric(
+        'g', 'desc', field_spec=[
+            monitor.StringField('name'),
+        ])
+    gauge.set(5)
+    self.assertIsInstance(gauge, monitor._MockMetric)
+
+
+class TestFlusherThread(unittest.TestCase):
+  """Sets up the flusher thread and mocks MetricServiceClient."""
+
+  @patch(
+      'clusterfuzz._internal.metrics.monitor.monitoring_v3.MetricServiceClient')
+  def test_flusher_thread(self, mock_client):
+    """Sets up the flusher thread and calls run()."""
+
+    monitor._monitoring_v3_client = mock_client.return_value
+    monitor._flusher_thread = monitor._FlusherThread()
+    monitor.FLUSH_INTERVAL_SECONDS = 1
+    monitoring_metrics.BOT_COUNT.set(1, {'revision': '1'})
+    monitoring_metrics.HOST_INCONSISTENT_COUNT.increment()
+    os.environ['BOT_NAME'] = 'bot-1'
+
+    monitor._flusher_thread.start()
+    mock_create_time_series = mock_client.return_value.create_time_series
+    while not mock_create_time_series.called:
+      pass
+    monitor._flusher_thread.stop()
+    time_series = mock_create_time_series.mock_calls[0][2]['time_series']
+    for i in range(1, len(time_series)):
+      self.assertLessEqual(time_series[i - 1].points[0].interval.start_time,
+                           time_series[i].points[0].interval.start_time)
```

Comment:
This comparison working suggests we don't need to manually construct a tuple of seconds and nanoseconds as the sort key - these timestamp objects are already comparable?

Reply:
Yes, timestamp objects are comparable. changed sorting condition accordingly.